src/libs/Navigation/linkTo.ts

```diff
@@ -89,12 +97,15 @@ export default function linkTo(navigation: NavigationContainerRef<RootStackParam
             action.type = CONST.NAVIGATION.ACTION_TYPE.REPLACE;
 
             // If this action is navigating to the RightModalNavigator and the last route on the root navigator is not RightModalNavigator then push
-        } else if (action.payload.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR && rootState?.routes?.at(-1)?.name !== NAVIGATORS.RIGHT_MODAL_NAVIGATOR) {
+        } else if (isModalNavigator(action.payload.name) && !isTargetNavigatorOnTop) {
+            if (isModalNavigator(topRouteName)) {
+                dismissModal('', navigation);
+            }
```

Comment:
Can you please clarify why is this needed?

Reply:
It's necessary to dismiss the already opened modal when we navigate to ModalNavigator. Before I removed the referral from the SearchPage, there had been possibility to navigate from LHP to RHP. In this case LHP should be dismissed before RHP is opened. It's the additional check for that.

Reply:
Furthermore, we need this check to properly handle keyboard shortcuts in the app. When we switch from the LHP to the RHP, the previous one should be closed. 

https://github.com/Expensify/App/assets/47774969/027cb2ac-63d7-4567-9d84-734dc1876790


https://github.com/Expensify/App/assets/47774969/b4117796-1f00-4c75-ad83-097f685d5f11

Reply:
Makes sense!